### PR TITLE
Make more `Selector` properties work on iOS

### LINKED ISSUE #1554: Disparity of `Selector` behavior between Android and iOS
Patrol has the [`Selector`](https://github.com/leancodepl/patrol/blob/d8e462d2f4418f9558f8bab7595e460f50042160/contracts.proto#L178-L191) and [`NativeView`](https://github.com/leancodepl/patrol/blob/d8e462d2f4418f9558f8bab7595e460f50042160/contracts.proto#L196-L206) classes. They provide cross-platform interaction with the underlying native UI.

Currently, there are many differences `Selector` and `NativeView` between its behavior on Android and iOS.

### More about these 2 classes

<details><summary>Selector</summary>

On Android, `Selector`s are translated to [`BySelector`](https://developer.android.com/reference/androidx/test/uiautomator/BySelector)s and/or [`UiSelector`](https://developer.android.com/reference/androidx/test/uiautomator/UiSelector).

On iOS, `Selector`s are translated to [`XCUIElementQuery`](https://developer.apple.com/documentation/xctest/xcuielementquery)-ies. Sometimes, we construct them using `NSSelector`. They allow for searching using many view properties: `identifier`, `title`, `label`, `value`, and `placeholderValue`.

</details> 

<details><summary>NativeView</summary>

On Android, `NativeView`s are created from [`UiObject2`](https://developer.android.com/reference/androidx/test/uiautomator/UiObject2)s.

On iOS, `NativeView`s are created from [`XCUIElement`](https://developer.apple.com/documentation/xctest/xcuielement)s.

</details> 

### Problems

- **on iOS, only `Selector.text` is taken into account** when searching for views
- some `Selector` fields, such as [`resourceId`](https://github.com/leancodepl/patrol/blob/d8e462d2f4418f9558f8bab7595e460f50042160/contracts.proto#L186), are Android-only. That's because the `Selector` API predates iOS support.
- some `NativeView` fields, such as [`resourceName`](https://github.com/leancodepl/patrol/blob/d8e462d2f4418f9558f8bab7595e460f50042160/contracts.proto#L203), are Android-only. Cause the same as above.
- #1403
- #663

### PR DESCRIPTION
**Pull request by bartekpacia**
This PR resolves #1554.

### CODE REVIEW

**Comment by bartekpacia** [comment]
Some thoughts for when I'm gone:

- I really think that `Selector.text` and `Selector.contentDescription` should be the same thing (preferably named `Selector.label`) @piotruela

**Comment by piotruela** [comment]
I made Selector's `textStartsWith` and `textContains` work with methods `tap`, `doubleTap`, `waitUntilVisible`, `enterText` and `getNativeViews`. I wrote some tests in `ios_selector_test.dart` but I spotted a weird behavior there. If I run just a single test it passes but when I try to run more than one at once they all fail. I could not find a cause of that but before we merge this PR we need to investigate this.

**Comment by jBorkowska** [comment]
I investigated failing tests, it is caused by another bug: https://github.com/leancodepl/patrol/issues/1868, I suggest to merge this PR regardless to those tests failing

**Comment by abhiche** [comment]
When do you plan to release this?

File: packages/patrol/darwin/Classes/AutomatorServer/Automator/Automator.swift [hunk 1]
```diff
@@ -1,3 +1,93 @@
+extension Selector {
```
Comment (zltnDC) [inline packages/patrol/darwin/Classes/AutomatorServer/Automator/Automator.swift:1]: This extension should be rather below #if PATROL_ENABLED.

**Review by piotruela — APPROVED** [review]

**Review by zltnDC — APPROVED** [review]

**Pull request merged**